x-pack/plugins/apm/public/components/alerting/ui_components/alerting_flyout/index.tsx

```diff
@@ -35,7 +36,7 @@ export function AlertingFlyout(props: Props) {
   const { start, end } = useTimeRange({ rangeFrom, rangeTo, optional: true });
 
   const environment =
-    'environment' in query ? query.environment : ENVIRONMENT_ALL.value;
+    'environment' in query ? query.environment! : ENVIRONMENT_ALL.value;
```

Comment:
I think we can improve this so we are extra careful, WDYT:
```
const queryEnvironment= 'environment' in query ? query.environment : undefined;
const environment = queryEnvironment || ENVIRONMENT_ALL.value;
```

This solves the cases:
1.  environment exists in query and is defined
2. environment exists in query and is not defined (empty)
3. environment doesn't exist